core/src/wsv.rs

```diff
@@ -335,15 +335,34 @@ impl WorldStateView {
     /// - If timestamp conversion to `u64` fails
     #[iroha_logger::log(skip_all, fields(block_height))]
     pub fn apply(&mut self, block: &VersionedCommittedBlock) -> Result<()> {
+        self.execute_transactions(block.as_v1())?;
+        debug!("All block transactions successfully executed");
+
+        self.apply_without_execution(block)?;
+
+        Ok(())
+    }
+
+    /// Apply transactions without actually executing them.
+    /// It's assumed that block's transaction was already executed (as part of validation for example).
```

Comment:
I wonder if it is possible to express this assumption via type system. Just a thought.

Reply:
Maybe we can use newtype pattern or something.